### DISCUSSION TITLE
rdk_doc/Advanced_development/linux_development/driver_development_x5/driver_gpio_dev/

### DISCUSSION THREAD

**Discussion by giscus** [D-Robotics/rdk_doc#52] [General]
# rdk_doc/Advanced_development/linux_development/driver_development_x5/driver_gpio_dev/

X5 芯片共有308个IO功能管脚，其中129个管脚都可以配置工作在gpio模式下，但是需要注意和其他功能管脚的复用关系。

https://d-robotics.github.io/rdk_doc/Advanced_development/linux_development/driver_development_x5/driver_gpio_dev/

**Comment by ss123-dev** [comment]
你好，请问一下我应该如何配置hobot-pps的驱动

**Reply by ss123-dev** [reply]
hobot_pps: hobot_pps {
		pinctrl-names = "default";
		pinctrl-0 = <&aon_gpio_6>;
		compatible = "hobot-pps";
		interrupt-controller;
		status = "okay";
		false_trigger_detect;
	};这是我添加的dts文件

**Reply by ss123-dev** [reply]
在x5.dtsi中添加

**Reply by ss123-dev** [reply]
X5 PIN SW Reg-V1.1.xlsx这个文件是否没有上传，我在上面链接处只看到了x3

**Reply by ss123-dev** [reply]
3.183201] horizon-lsio-pinctrl 34180000.lsio_iomuxc: pin lsio_spi2_ssn already requested by gpiochip4; cannot claim for hobot_pps
[    3.195051] horizon-lsio-pinctrl 34180000.lsio_iomuxc: pin-21 (hobot_pps) status -22
[    3.216487] hobot-pps hobot_pps: Error applying setting, reverse things back
如何取消gpiochip4的占用

**Reply by WuChao-2024** [reply]
你好，X5的驱动开发指南在手册中已经更新，欢迎体验。
